sktime/utils/parallel.py

```diff
@@ -0,0 +1,97 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Common abstraction utilities for parallelization backends.
+
+New parallelization or iteration backends can be added easily as follows:
+
+* Add a new backend name to ``backend_dict``, syntax is
+  backend_name: backend_type, where backend_type collects backend options,
+  e.g., multiple options for a single parallelization backend.
+* Add a new function to ``para_dict``, should have name
+  ``_parallelize_<backend_name>`` and take the same arguments as
+  ``_parallelize_none``. Ensure that ``backend`` is an argument,
+  even if there is only one backend option for the backend_type
+* add the backend string in the docstring of parallelize, and any downstream
+  functions that use ``parallelize`` and expose the backend parameter an argument
+"""
+
+__author__ = ["fkiraly"]
+
+
+def parallelize(fun, iter, meta=None, backend=None):
```

Comment:
A bit nickpiking, but if you dont mind. 

For readability (also common practices and conventions), I suppose having the private utility functions before the interface function `parallelize` that uses them would help other readers encounter the utility functions before seeing how they are used.

I would suggest having them ordered as the following 
1. capitalised backend_dict (as it is a constant), 
2. private functions
3. capitalised para_dict 
4. interface function

Reply:
> that uses them would help other readers encounter the utility functions before seeing how they are used.

interesting - I have intentionally done it the opposite way round, so readers encounter the public function which is "at the top" of the module interface before they encounter the entrails... following the general best practice recommendation that a review of a code base should proceed top-down (rather than bottom-up)